doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,255 @@
+# We take advantage of the ``animation_frame`` parameter in
+# Plotly's ``imshow`` function. As a side note, when the
+# ``binary_string`` parameter is set to ``True``, the image is
+# represented as grayscale.
+
+fig = px.imshow(
+    image_seq[12:22, :, :],
+    animation_frame=0,
+    height=500,
+    width=500,
+    binary_string=True,
+    labels={'animation_frame': 'time point'},
+    title='Sample of in-vivo human cornea'
+)
+plotly.io.show(fig)
+
+#####################################################################
+# Average over time
+# =================
+# First, we want to detect those dark spots where the data are lost. In
+# technical terms, we want to *segment* the dark spots (for
+# all frames in the sequence). Unlike the actual data (signal), the dark spots
+# do not move from one frame to the next; they are still. Therefore, we begin
+# by computing the time average of the image sequence. We shall use this
+# time-averaged image to segment the dark spots, the latter then standing out
+# with respect to the background (blurred signal).
+
+image_avg = np.mean(image_seq, axis=0)
+
+print(f'shape: {image_avg.shape}')
+
+fig, ax = plt.subplots(figsize=(6, 6))
+
+ax.imshow(image_avg, cmap='gray')
+ax.set_title('Time-averaged image')
+
+fig.tight_layout()
+
+#####################################################################
+# Use local thresholding
+# ======================
+# To segment the dark spots, we use thresholding. The images we are working
+# with have uneven illumination, which causes variations in the (absolute)
+# intensities of the foreground and the background, from one region to another
+# (distant) one. It is therefore more fitting to compute different threshold
+# values across the image, one for each region. This is called adaptive (or
+# local) thresholding, as opposed to the usual thresholding procedure which
+# employs a single (global) threshold for all pixels in the image.
+#
+# When calling the ``threshold_local`` function from the ``filters`` module,
+# we may change the default neighborhood size (``block_size``), i.e., the
+# typical size (number of pixels) over which illumination varies,
+# as well as the ``offset`` (shifting the neighborhood's weighted mean).
+# Let us try two different values for ``block_size``:
+
+thresh_1 = filters.threshold_local(image_avg, block_size=21, offset=15)
+thresh_2 = filters.threshold_local(image_avg, block_size=43, offset=15)
+
+mask_1 = image_avg < thresh_1
+mask_2 = image_avg < thresh_2
+
+#####################################################################
+# Let us define a convenience function to display two plots side by side, so
+# it is easier for us to compare them:
+
+def plot_comparison(plot1, plot2, title1, title2):
+    fig, (ax1, ax2) = plt.subplots(
+        ncols=2,
+        figsize=(12, 6),
+        sharex=True,
+        sharey=True
+    )
+    ax1.imshow(plot1, cmap='gray')
+    ax1.set_title(title1)
+    ax2.imshow(plot2, cmap='gray')
```

Comment:
Has there been a discussion about the colormap to use through most of the example? "gray" is not a bad choice here in my opinion but if I where to do this exploration on my own I would probably use a colormap with colors, because it helps with separating values even more. E.g. "cividis" would be a colorblind friendly map.

Reply:
Do you mean *viridis*? Why not but, in this case, there are only *two* colors, so 'gray' really means black and white -- `plot_comparison()` is only used to visualize *masks* in this tutorial.

Reply:
Ha, true, there's only masks here. So actual color wouldn't make much of a difference. :sweat_smile: Then it's probably best to just stay with gray from a pedagogical stand point. :+1:  

I actually meant "cividis" because it's supposed to be more colorblind friendly (see [this paper](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0199239) found as a reference [here](https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html)).

Reply:
Ah, thanks for the references! I didn't remember of `cividis` (if I ever actually came across it in the past). As an aside, I'm a big fan of all the colormap work which has been done in the oceanography community (thinking of @kthyng :heart:).

Reply:
So fun to get this ping! Thanks!!

Reply:
That got me curious. I assume you are referencing https://matplotlib.org/cmocean/? That really looks like something to keep in the back of my mind. :D

Reply:
`cmocean` is beautiful, and I love that it is tuned to different applications! FWIW, viridis and the other perceptually uniform color maps work quite well for red-green colorblind deficiency. The paper Lars cites above is quite transparent in its comparisons, and writes re Viridis:

> "While its CVD-simulated counterpart is close to optimal..."

Also see http://bids.github.io/colormap/ for examples of how the viscm tool (which just had a new release, thanks to @mfisher87!) does colorblind simulations of viridis, magma, and plasma.

In the end, the choice between viridis and cividis comes down to a trade off between colorfulness and close-to-perfect colorblind representation. And I don't think anyone will fault you, whichever way you choose.